### ISSUE TITLE
If tests from multiple test suites are being run in parallel, try build may look at the wrong results

### ISSUE THREAD

**Issue by nagisa** [dtolnay/trybuild#93]
Consider code like this:

```rust
// src/tests/a.rs
#[test]
fn a() {
    let t = trybuild::TestCases::new();
    t.compile_fail("a_case.rs");
}

// src/tests/b.rs
#[test]
fn b() {
    let t = trybuild::TestCases::new();
    t.compile_fail("b_case.rs");
}
```

If these two test suites run in parallel, `trybuild` will clobber (AFAICT) the same temporary crate directory (in `target`) and compare outputs from wrong crate build.

`cargo` itself does not run test suites in parallel, however, so reproducing this may involve running `cargo test` first, and then running the produced test suites by hand, both at the same time, or perhaps running multiple instances of `cargo` at once.

My understanding of the root cause may be wrong, but if it is correct, I think it could be trivially fixed by "just" making sure the generated `-test` crates are unique.

**Comment by nagisa** [comment]
Looks like a duplicate of #58
